utils/badges.js

```diff
@@ -0,0 +1,91 @@
+// @ts-check
+
+// Gets the badges that belong to the user based on a list of roles.
+
+const u = require("../utils/utils"),
+  Discord = require("discord.js"),
+  config = require("../config/config.json"),
+  gs = require("google-spreadsheet"),
+  fs = require("fs");
+
+/**
+ * Gets all badge data from the Google Sheet.
+ * @returns {Promise<Map>}
+ */
+async function getBadgeData() {
+  if (!config.google.sheets.config) {
+    console.log("No Sheets ID");
+    return new Map();
+  }
+  const doc = new gs.GoogleSpreadsheet(config.google.sheets.config);
+  try {
+    await doc.useServiceAccountAuth(config.google.creds);
+    await doc.loadInfo();
+    // @ts-ignore sheets stuff
+    const roles = await doc.sheetsByTitle["Roles"].getRows();
+
+    const badgeMap = new Map();
+
+    for (const role of roles) {
+      // Only add to the map if they have a role ID...
+      if (!role["Base Role ID"] || role["Base Role ID"] == '') continue;
+      // ...and if they have a badge listed...
+      if (!role["Badge"] || role["Badge"] == "") continue;
+      // ...and if the badge path is valid.
+      if (!fs.existsSync(`./media/badges/${role["Badge"]}.png`)) continue;
+
+      badgeMap.set(role["Base Role ID"], {
+        title: role["Role Reference"],
+        image: `${role["Badge"]}.png`,
+        // if there are lower roles, split them, and then remove the one at the end that's just an empty string.
+        overrides: (role["Lower Roles"] ? role["Lower Roles"].split(" ").filter((x) => x) : null)
```

Comment:
for type simplicity it should probably always return an array. You can test for length instead of null.

Reply:
Changed it to an array. Also changed another line that checked to see if obj.overrides existed. Forgot that an empty array could just do that